addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -1179,6 +1178,115 @@ QUnit.test('open 2 chat windows: check shift operations are available', async fu
     );
 });
 
+QUnit.test('open & fold 2 chat windows: check shift operations are available', async function (assert) {
+    assert.expect(9);
+
+    const channel = {
+        channel_type: "channel",
+        id: 20,
+        is_pinned: true,
+        name: "General",
+    };
+    const chat = {
+        channel_type: "chat",
+        direct_partner: [{
+            id: 7,
+            name: "Demo",
+        }],
+        id: 10,
+        is_pinned: true,
+    };
+    this.data['mail.channel'].records = [channel, chat];
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [channel],
+            channel_direct_message: [chat],
+        },
+    });
+    await this.start();
+    await afterNextRender(() => {
+        document.querySelector('.o_MessagingMenu_toggler').click();
+    });
+    await afterNextRender(() => {
+        document.querySelectorAll('.o_MessagingMenu_dropdownMenu .o_NotificationList_preview')[0].click();
+    });
+    await afterNextRender(() => {
+        document.querySelector('.o_MessagingMenu_toggler').click();
+    });
+    await afterNextRender(() => {
+        document.querySelectorAll('.o_MessagingMenu_dropdownMenu .o_NotificationList_preview')[1].click();
+    });
+
+    assert.containsN(
+        document.body,
+        '.o_ChatWindow',
+        2,
+        "should have opened 2 chat windows"
+    );
+    await afterNextRender(() => {
+        document.querySelectorAll('.o_ChatWindowHeader')[0].click();
```

Comment:
We should avoid relying on index in `querySelectorAll()`, because order in DOM is quite arbitrary (same with UI, especially with chat windows with is affected by locale text direction),

- for previews: make use of `data-thread-local-id` to select appropriate thread (! don't leak local id value !)
- for chat windows: make use of `data-visible-index` like test `'chat window: TAB cycle with 3 open chat windows'` (just this test use index, other tests have not been converted yet...)

Reply:
Just copy/pasted the "normal" test of shifting left/right.
In fact, I don't care here of the order, I just wanna check you can shift left/right a folded chat window

Reply:
The order and the locale (LTR/RTL) are important for checking the result of shift left/right. Also the presence of these buttons depends on both.

Reply:
I went to a solution where order does not really matter